diff --git a/CMakeLists.txt b/CMakeLists.txt
index b44ef8b..85c29cd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -173,16 +173,20 @@
 
 IF(NOT ANDROID)
 IF(APPLE)
-    # Determine the canonical name of the selected Platform SDK
-    EXECUTE_PROCESS(COMMAND "/usr/bin/sw_vers" "-productVersion"
-                    OUTPUT_VARIABLE OSG_OSX_SDK_NAME
-                    OUTPUT_STRIP_TRAILING_WHITESPACE)
-    STRING(REPLACE "." ";" MACOS_VERSION_LIST ${OSG_OSX_SDK_NAME})
-    LIST(GET MACOS_VERSION_LIST 0 MACOS_VERSION_MAJOR)
-    LIST(GET MACOS_VERSION_LIST 1 MACOS_VERSION_MINOR)
-    LIST(GET MACOS_VERSION_LIST 2 MACOS_VERSION_PATCH)
+    IF(CMAKE_OSX_SYSROOT)
+        SET(OSG_OSX_SDK_NAME "${CMAKE_OSX_SYSROOT}")
+    ELSE()
+        # Determine the canonical name of the selected Platform SDK
+        EXECUTE_PROCESS(COMMAND "/usr/bin/sw_vers" "-productVersion"
+                        OUTPUT_VARIABLE OSG_OSX_SDK_NAME
+                        OUTPUT_STRIP_TRAILING_WHITESPACE)
+        STRING(REPLACE "." ";" MACOS_VERSION_LIST ${OSG_OSX_SDK_NAME})
+        LIST(GET MACOS_VERSION_LIST 0 MACOS_VERSION_MAJOR)
+        LIST(GET MACOS_VERSION_LIST 1 MACOS_VERSION_MINOR)
+        LIST(GET MACOS_VERSION_LIST 2 MACOS_VERSION_PATCH)
 
-    SET(OSG_OSX_SDK_NAME "macosx${MACOS_VERSION_MAJOR}.${MACOS_VERSION_MINOR}")
+        SET(OSG_OSX_SDK_NAME "macosx${MACOS_VERSION_MAJOR}.${MACOS_VERSION_MINOR}")
+    ENDIF()
 
     # Trying to get CMake to generate an XCode IPhone project, current efforts are to get iphoneos sdk 3.1 working
     # Added option which needs manually setting to select the IPhone SDK for building. We can only have one of the below
@@ -799,6 +799,8 @@ IF(NOT ANDROID)
         ENDIF()
 
     ELSE()
+        FIND_PACKAGE(JPEG)
+        FIND_PACKAGE(PNG)
         FIND_PACKAGE(TIFF)
         FIND_PACKAGE(QuickTime)
         FIND_PACKAGE(QTKit)
@@ -1108,7 +1110,7 @@ IF(APPLE AND NOT ANDROID)
         # FORCE is used because the options are not reflected in the UI otherwise.
         # Seems like a good place to add version specific compiler flags too.
         IF(NOT OSG_CONFIG_HAS_BEEN_RUN_BEFORE)
-            IF(${OSG_OSX_SDK_NAME} STREQUAL "macosx10.8" OR ${OSG_OSX_SDK_NAME} STREQUAL "macosx10.9" OR ${OSG_OSX_SDK_NAME} STREQUAL "macosx10.10")
+            IF(${OSG_OSX_SDK_NAME} STREQUAL "macosx10.8" OR ${OSG_OSX_SDK_NAME} STREQUAL "macosx10.9" OR ${OSG_OSX_SDK_NAME} STREQUAL "macosx10.10" OR ${OSG_OSX_SDK_NAME} STREQUAL "macosx10.11")
                 SET(OSG_DEFAULT_IMAGE_PLUGIN_FOR_OSX "imageio" CACHE STRING "Forced imageio default image plugin for OSX" FORCE)
                 # 64 Bit Works, i386,ppc is not supported any more
                 SET(CMAKE_OSX_ARCHITECTURES "x86_64" CACHE STRING "Build architectures for OSX" FORCE)
diff --git a/CMakeModules/FindQuickTime.cmake b/CMakeModules/FindQuickTime.cmake
index bffc99b..be7a6a4 100644
--- a/CMakeModules/FindQuickTime.cmake
+++ b/CMakeModules/FindQuickTime.cmake
@@ -65,7 +65,7 @@ ELSE()
       ENDIF()
       # Disable quicktime for >= 10.7, as it's officially deprecated
 
-      IF(${OSG_OSX_SDK_NAME} STREQUAL "macosx10.7" OR ${OSG_OSX_SDK_NAME} STREQUAL "macosx10.8" OR ${OSG_OSX_SDK_NAME} STREQUAL "macosx10.9" OR ${OSG_OSX_SDK_NAME} STREQUAL "macosx10.10")
+      IF(${OSG_OSX_SDK_NAME} STREQUAL "macosx10.7" OR ${OSG_OSX_SDK_NAME} STREQUAL "macosx10.8" OR ${OSG_OSX_SDK_NAME} STREQUAL "macosx10.9" OR ${OSG_OSX_SDK_NAME} STREQUAL "macosx10.10" OR ${OSG_OSX_SDK_NAME} STREQUAL "macosx10.11")
           MESSAGE("disabling quicktime because it's not supported by the selected SDK ${OSG_OSX_SDK_NAME}")
           SET(QUICKTIME_FOUND "NO")
       ENDIF()
diff --git a/src/osgDB/Registry.cpp b/src/osgDB/Registry.cpp
index 14a8d04..83bacf2 100644
--- a/src/osgDB/Registry.cpp
+++ b/src/osgDB/Registry.cpp
@@ -308,18 +308,6 @@ Registry::Registry()
 
     addFileExtensionAlias("js", "V8");
 
-#if defined(DARWIN_IMAGEIO)
-    addFileExtensionAlias("jpg",  "imageio");
-    addFileExtensionAlias("jpe",  "imageio");
-    addFileExtensionAlias("jpeg", "imageio");
-    addFileExtensionAlias("tif",  "imageio");
-    addFileExtensionAlias("tiff", "imageio");
-    addFileExtensionAlias("gif",  "imageio");
-    addFileExtensionAlias("png",  "imageio");
-    addFileExtensionAlias("psd",  "imageio");
-    addFileExtensionAlias("tga",  "imageio");
-
-#endif
 
 #if defined(USE_QTKIT)
     addFileExtensionAlias("mov",  "QTKit");
diff --git a/src/osgPlugins/CMakeLists.txt b/src/osgPlugins/CMakeLists.txt
index f2ca656..dbeb39c 100644
--- a/src/osgPlugins/CMakeLists.txt
+++ b/src/osgPlugins/CMakeLists.txt
@@ -239,9 +239,6 @@ ENDIF()
 #    ADD_SUBDIRECTORY(OpenAL)
 # ENDIF()
 
-IF(APPLE AND NOT ANDROID)
-    ADD_PLUGIN_DIRECTORY(imageio)
-ENDIF()
 
 IF(AV_FOUNDATION_FOUND)
     ADD_PLUGIN_DIRECTORY(avfoundation)
